joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala

```diff
@@ -42,20 +43,34 @@ class Py2CpgOnFileSystem extends X2CpgFrontend[Py2CpgOnFileSystemConfig] {
         } else {
           Nil
         }
-      val inputFiles = collectInputFiles(config.inputDir, ignorePrefixes)
-      val inputProviders = inputFiles.map { inputFile => () =>
+      val inputFiles = collectInputFiles(config.inputDir, ignorePrefixes, config.requirementsTxt)
+      val inputProviders = inputFiles.filter(_.endsWith("py")).map { inputFile => () =>
         {
           val content = IOUtils.readLinesInFile(inputFile).mkString("\n")
           Py2Cpg.InputPair(content, inputFile.toString, config.inputDir.relativize(inputFile).toString)
         }
       }
-
-      val py2Cpg = new Py2Cpg(inputProviders, cpg)
+      val configInputProviders =
+        inputFiles
+          .filter { inputFile =>
+            inputFile.endsWith(config.requirementsTxt)
+          }
+          .map { inputFile => () =>
+            {
+              val content = IOUtils.readLinesInFile(inputFile).mkString("\n")
+              Py2Cpg.InputPair(content, inputFile.toString, config.inputDir.relativize(inputFile).toString)
+            }
+          }
+      val py2Cpg = new Py2Cpg(inputProviders, configInputProviders, cpg)
```

Comment:
Instead of filtering the inputFiles i would suggest `collectInputFiles` returns a tuple of python input files and the requirements.txt.